### PR TITLE
Add ability to disable creation of public projects and xforms

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- Added new `ALLOW_PUBLIC_DATASETS` settings
- Added ability to disable creation of public projects and forms

### CODE REVIEW

File: onadata/libs/serializers/xform_serializer.py [hunk 1]
```diff
@@ -328,6 +328,28 @@ def validate_public_key(self, value):  # pylint: disable=no-self-use
                 _('The public key is not a valid base64 RSA key'))
         return clean_public_key(value)
 
+    def _check_if_allowed_public(self, value):
```
Comment (pld) [inline onadata/libs/serializers/xform_serializer.py:331]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5585241)

File: onadata/libs/serializers/project_serializer.py [hunk 2]
```diff
@@ -369,6 +369,15 @@ def validate(self, attrs):
             })
         return attrs
 
+    def validate_public(self, value):
```
Comment (pld) [inline onadata/libs/serializers/project_serializer.py:372]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5585241)

File: onadata/libs/serializers/project_serializer.py [hunk 3]
```diff
@@ -369,6 +369,15 @@ def validate(self, attrs):
             })
         return attrs
 
+    def validate_public(self, value):  # pylint: disable=no-self-use
+        """
+        Validate the public field
+        """
+        if not settings.ALLOW_PUBLIC_DATASETS and value:
+            raise serializers.ValidationError(
+                'Public projects are currently disabled.')
```
Comment (ukanga) [inline onadata/libs/serializers/project_serializer.py:378]: Can you please enable translation support? 
```python
_("Public projects are currently disabled.")
```

Comment (DavisRayM) [reply onadata/libs/serializers/project_serializer.py:378]: Yes, let me change that

**Pull request merged**